lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateTaskAccTest.java

```diff
@@ -0,0 +1,582 @@
+          assertThatThrownBy(() -> taskService.updateTask(task))
+              .isInstanceOf(InvalidArgumentException.class);
+        };
+    return DynamicTest.stream(list.iterator(), Pair::getLeft, test);
+  }
+
+  @WithAccessId(user = "user-taskrouter")
+  @Test
+  void should_ThrowException_When_UserIsNotAuthorized() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    task.setPrimaryObjRef(
+        taskService.newObjectReference("company", "system", "instance", "value", "type"));
+    ClassificationSummary classificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "admin");
+    task.setClassificationKey(classificationSummary.getKey());
+
+    ThrowingCallable call = () -> taskService.updateTask(task);
+
+    NotAuthorizedOnWorkbasketException e =
+        catchThrowableOfType(call, NotAuthorizedOnWorkbasketException.class);
+    assertThat(e.getCurrentUserId()).isEqualTo("user-taskrouter");
+    assertThat(e.getWorkbasketId()).isEqualTo(defaultWorkbasketSummary.getId());
+    assertThat(e.getRequiredPermissions()).containsExactly(WorkbasketPermission.READ);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_ThrowException_When_TaskHasAlreadyBeenUpdated() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    final Task task2 = taskService.getTask(task.getId());
+
+    task.setCustomField(CUSTOM_1, "willi");
+    Thread.sleep(10);
+    taskService.updateTask(task);
+    task2.setCustomField(CUSTOM_2, "Walter");
+    // TODO flaky test ... if speed is too high,
+    assertThatThrownBy(() -> taskService.updateTask(task2))
+        .isInstanceOf(ConcurrencyException.class)
+        .hasMessage(
+            "The entity with id '%s' cannot be updated since it has been modified while editing.",
+            task.getId());
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_UpdateTask_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
+      throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    ClassificationSummary classificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "admin");
+    task.setClassificationKey(classificationSummary.getKey());
```

Comment:
Please move setting attributes for update (like "setClassificationKey") in the "When" part of the test. Could you do it for other tests too?